sunpy/map/mapbase.py

```diff
@@ -2819,7 +2832,41 @@ def plot(self, *, annotate=True, axes=None, title=True, autoalign=False,
         else:
             data = np.ma.array(np.asarray(self.data), mask=self.mask)
 
-        if autoalign == 'pcolormesh':
+        if autoalign == 'image':
+            ny, nx = self.data.shape
+            pixel_perimeter = grid_perimeter(nx, ny) - 0.5
+
+            transform = axes.get_transform(self.wcs) - axes.transData
+            data_perimeter = transform.transform(pixel_perimeter)
+
+            if not np.all(np.isfinite(data_perimeter)):
+                raise RuntimeError("Cannot draw an autoaligned image at all due to its coordinates. "
+                                   "Try specifying autoalign=mesh.")
+
+            min_x, min_y = np.min(data_perimeter, axis=0)
+            max_x, max_y = np.max(data_perimeter, axis=0)
+
+            data_corners = data_perimeter[[0, nx, nx + ny, 2*nx + ny], :]
+            if not (np.allclose([min_x, min_y], np.min(data_corners, axis=0))
+                    and np.allclose([max_x, max_y], np.max(data_corners, axis=0))):
+                warn_user("Cannot draw all of the autoaligned image due to the warping required. "
+                          "Specifying autoalign=mesh is recommended.")
+
+            # Draw the image, but revert to the prior data limits because matplotlib does not account for the transform
+            old_datalim = copy.deepcopy(axes.dataLim)
+            ret = axes.imshow(data, transform=transform + axes.transData, **imshow_args)
+            axes.dataLim = old_datalim
+
+            # Update the data limits based on the transformed perimeter
+            ret.sticky_edges.x[:] = [min_x, max_x]
+            ret.sticky_edges.y[:] = [min_y, max_y]
+            axes.update_datalim([(min_x, min_y), (max_x, max_y)])
+            axes._request_autoscale_view()
```

Comment:
What's this hidden method call doing here :wink: it's not a deal breaker, but maybe a comment or something about why we are using it in case something breaks upstream.